### PR TITLE
metrics: add server info metric

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Add PD server info in Prometheus metrics. This is use for DBaaS metrics.

related PR: https://github.com/pingcap/tidb/pull/22556

### What is changed and how it works?

N/A

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
```shell
# start the pd-server first, also start the prometheus server
▶ bin/pd-server
# query the metrics info:
▶ curl '127.0.0.1:9090/api/v1/query?query=pd_server_info&time=2021-01-27T06:15:00.781Z'
{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"pd_server_info","hash":"cbb89ec085979727cf1250aef92fe3c6ff4e63ee","instance":"127.0.0.1:2379","job":"pd","version":"v4.0.0-rc.2-434-gcbb89ec0-dirty"},"value":[1611728100.781,"1611728088"]}]}}%
```

Side effects

- N/A

### Release note 

- N/A

### CODE REVIEW

**Review by breeswish — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

File: server/metrics.go [hunk 1]
```diff
@@ -83,6 +83,14 @@ var (
 			Help:      "Bucketed histogram of processing time (s) of handled store heartbeat requests.",
 			Buckets:   prometheus.ExponentialBuckets(1, 2, 12),
 		}, []string{"address", "store"})
+
+	serverInfo = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "pd",
+			Subsystem: "server",
+			Name:      "info",
+			Help:      "Indicate the pd server info, and the value is the start timestamp (s).",
```
Comment (iosmanthus) [inline server/metrics.go:92]: Maybe modifying to `the server startup timestamp?` will be less confused?

**Review by rleungx — APPROVED** [review]

**Pull request merged**